graph/pull.go

```diff
@@ -28,10 +18,24 @@ type ImagePullConfig struct {
 	OutStream   io.Writer
 }
 
+type Puller interface {
+	Pull(image, name, tag string, endpoint registry.APIEndpoint, imagePullConfig *ImagePullConfig, sf *streamformatter.StreamFormatter) (fallback bool, err error)
+}
+
+func (s *TagStore) NewPuller(endpoint registry.APIEndpoint, repoInfo *registry.RepositoryInfo) (Puller, error) {
```

Comment:
@RichardScothern oh you mean i could pass it as a param to a `NewPuller` function, instead of a method?